pkg/ingester/client/streaming.go

```diff
@@ -132,11 +132,7 @@ func (s *SeriesChunksStreamReader) StartBuffering() {
 				}
 			}
 
-			if err := s.queryLimiter.AddChunks(chunkCount); err != nil {
-				onError(err)
-				return
-			}
-
+			// The chunk count limit is enforced earlier, while we're reading series labels, so we don't need to do that here.
```

Comment:
It seems like you can remove the intermediate `chunkCount` and always increment `totalChunks`

Reply:
Good point, addressed in https://github.com/grafana/mimir/pull/5447